src/index.js

```diff
@@ -34,33 +33,25 @@ class IpfsRepo {
   /**
    * @param {string} repoPath - path where the repo is stored
    * @param {object} options - Configuration
-   * @param {Datastore} options.fs
-   * @param {Leveldown} options.level
-   * @param {object} [options.fsOptions={}]
+   * @param {datastore} options.blockStore
+   * @param {datastore} options.dataStore
+   * @param {object} [options.blockStoreOptions={}]
+   * @param {object} [options.dataStoreOptions={}]
    * @param {bool} [options.sharding=true] - Enable sharding (flatfs on disk), not needed in the browser.
    * @param {string} [options.lock='fs'] - Either `fs` or `memory`.
    */
   constructor (repoPath, options) {
     assert.equal(typeof repoPath, 'string', 'missing repoPath')
 
-    if (options == null) {
-      options = require('./default-options')
-    }
-
+    const defaultOptions = require('./default-options')
     this.closed = true
     this.path = repoPath
-    this.options = Object.assign({
-      sharding: true,
-      lock: 'fs'
-    }, options)
-    this._fsOptions = Object.assign({}, options.fsOptions)
-    const FsStore = this.options.fs
-    this._fsStore = new FsStore(this.path, Object.assign({}, this._fsOptions, {
-      extension: ''
-    }))
+    this.options = Object.assign({ lock: 'memory', sharding: true }, options || defaultOptions)
```

Comment:
Question about this. The `fs` lock appears to not use the datastore but to use fs api's directly. Supposing you were going to use a virtual file storage, and not the local file system, such as s3 would you still want a file system lock? It's not that the file system is completely off limits from writing to but the machine can get taken down and back up and the file system is ephemeral with the instance. Should the lock file also be stored on external storage or is it tied to the life cycle of the process?

Is its purpose to prevent other apps from writing the same files and if so shouldn't the lock actually in the one of the stores?